scheduler/plugin/guest.go

```diff
@@ -75,14 +82,26 @@ func (pl *wasmPlugin) newGuest(ctx context.Context) (*guest, error) {
 	return &guest{
 		guest:       g,
 		out:         &out,
+		enqueueFn:   g.ExportedFunction(guestExportEnqueue),
 		prefilterFn: g.ExportedFunction(guestExportPreFilter),
 		filterFn:    g.ExportedFunction(guestExportFilter),
 		scoreFn:     g.ExportedFunction(guestExportScore),
 		callStack:   callStack,
 	}, nil
 }
 
-// PreFilter calls guestExportPreFilter.
+// eventsToRegister calls guestExportEnqueue.
+func (g *guest) eventsToRegister(ctx context.Context) []framework.ClusterEvent {
+	defer g.out.Reset()
+	callStack := g.callStack
+	if err := g.enqueueFn.CallWithStack(ctx, callStack); err != nil {
+		// framework.EnqueueExtensions.EventsToRegister() does not return an error
+		return nil
```

Comment:
can we panic() instead (for now until https://github.com/kubernetes/kubernetes/issues/119323 is done)?
EventToRegister is only called when the scheduler is being initialized so it makes more sense to me to panic() in error cases so that the users can notice their mistake or something by panic

Reply:
good idea I can panic here